src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java

```diff
@@ -685,15 +631,51 @@ private void updateLatestRealtimeTask(
             );
     }
 
-    private void releaseLock(AnomalyDetectorJob jobParameter, LockService lockService, LockModel lock) {
-        lockService
-            .release(
-                lock,
-                ActionListener
-                    .wrap(
-                        released -> { log.info("Released lock for AD job {}", jobParameter.getName()); },
-                        exception -> { log.error("Failed to release lock for AD job: " + jobParameter.getName(), exception); }
-                    )
-            );
+    private LockModel acquireLock(JobExecutionContext context, Long lockDurationSeconds) throws Exception {
+
+        // Build request body
+        AcquireLockRequest acquireLockRequestBody = new AcquireLockRequest(
+            context.getJobId(),
+            context.getJobIndexName(),
+            lockDurationSeconds
+        );
+
+        // Create acquire lock request
+        Request acquireLockRequest = new Request("GET", String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_lock"));
+        acquireLockRequest
+            .setJsonEntity(Strings.toString(acquireLockRequestBody.toXContent(JsonXContent.contentBuilder(), ToXContent.EMPTY_PARAMS)));
+
+        // Parse response map fields for lock model
+        Response acquireLockResponse = client.performRequest(acquireLockRequest);
+        XContentParser parser = XContentType.JSON
+            .xContent()
+            .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, acquireLockResponse.getEntity().getContent());
+
+        AcquireLockResponse acquireLockResponseBody = AcquireLockResponse.parse(parser);
+        return acquireLockResponseBody.getLock();
+    }
+
+    private void releaseLock(AnomalyDetectorJob jobParameter, LockModel lock) {
+
+        Request releaseLockRequest = new Request(
+            "PUT",
+            String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_release_lock")
+        );
+        releaseLockRequest.addParameter(LockModel.LOCK_ID, lock.getLockId());
+
+        try {
+            Response releaseLockResponse = client.performRequest(releaseLockRequest);
```

Comment:
Is `performRequest` async?

Reply:
`performRequest ` is a blocking call. There is currently no async mechanism to send a rest call via the SDKRestClient.

Reply:
@joshpalis let's add async in SDK and use it here. This can be done in the next PR.